modules/lmdbbackend/lmdbbackend.cc

```diff
@@ -53,8 +54,14 @@
 // List the class version here. Default is 0
 BOOST_CLASS_VERSION(LMDBBackend::KeyDataDB, 1)
 
+static ReadWriteLock s_lmdbOpenUpgradeLock;
+
 LMDBBackend::LMDBBackend(const std::string& suffix)
 {
+  // we lock to avoid a race condition when we do a schema upgrade during startup
+  // non-upgrade startups should be very cheap so this lock should not hurt performance
+  WriteLock l(&s_lmdbOpenUpgradeLock);
```

Comment:
Just a nit but if we don't have a lot of concurrent readers a simple `std::mutex` would be faster :)

Reply:
ah thanks!